Pygrib install fails with build error

The command `conda env create --file condaEnv/manyFEMSenv.yml` fails with the following error:

```
 #warning "Using deprecated NumPy API, disable it with " \
       ^
      src/pygrib/_pygrib.c:756:10: fatal error: 'grib_api.h' file not found
      #include "grib_api.h"
               ^~~~~~~~~~~~
      1 warning and 1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
```

Error can be fixed/bypassed by moving pygrib from the `pip` install dependencies (line 27 of [condaEnv/manyFEMSenv.yml](https://github.com/DurhamARC/ManyFEWS/blob/main/condaEnv/manyFEMSenv.yml#L27)) into the conda dependencies list instead.

Unless anyone objects, I'll commit this change.

Reference issue: https://github.com/jswhit/pygrib/issues/115

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,60 @@
+# ----------------------------------------------------------------------------
+# Build node / Frontend assets
+FROM node:alpine3.15 as build_node
+
+# Based on https://cli.vuejs.org/guide/deployment.html#docker-nginx
+WORKDIR /app
+
+# Install Python (required for node-gyp)
+RUN apk add --update python3 make g++ && \
+    rm -rf /var/cache/apk/*
+
+# Install app dependencies
+COPY manyfews/package.json .
+RUN npm install
+
+# Copy files and build app
+COPY manyfews/ .
+RUN npm run build
+
+
+# ----------------------------------------------------------------------------
+# Deployment stage
+FROM continuumio/miniconda3:4.12.0 as deploy
+MAINTAINER Samantha Finnigan <samantha.finnigan@durham.ac.uk>, ARC Durham University
+
+WORKDIR /app
+
+# set environment variables (don't buffer stdout, don't write bytecode)
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# Add a celery user to avoid running as root
+RUN mkdir -p /var/log/celery/ /var/run/celery/ &&\
+    useradd -G root celery && \
+    chgrp -Rf root /var/log/celery/ /var/run/celery/ && \
+    chmod -Rf g+w /var/log/celery/ /var/run/celery/ && \
+    chmod g+w /etc/passwd
+
+# https://pythonspeed.com/articles/conda-docker-image-size/
+# Create the environment:
+COPY condaEnv/ .
+RUN conda env create -f manyFEMSenv.yml
+
+# Make RUN commands use the new environment:
+RUN echo "conda activate ManyFEWS" >> ~/.bashrc
+SHELL ["/bin/bash", "--login", "-c"]
+
+# Demonstrate the environment is activated:
+RUN echo "Make sure django is installed:"
+RUN python -c "import django"
+
+# The code to run when container is started:
+COPY entrypoint.sh ./
+COPY --from=build_node /app .
+COPY Data ./
+
+RUN chmod +x entrypoint.sh
+ENTRYPOINT ["./entrypoint.sh"]
+EXPOSE 8000
+CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]

--- a/condaEnv/manyFEMSenv.yml
+++ b/condaEnv/manyFEMSenv.yml
@@ -19,11 +19,12 @@ dependencies:
   - pytest
   - retrying
   - selenium
+  - pygrib
+  - gdal
   - pip:
     - django-geojson==3.2.0
     - django-environ
     - django-phonenumber-field[phonenumberslite]
     - git+https://github.com/mt-climate-office/Zentra-API
-    - pygrib
     - tqdm
 prefix: /usr/local/Caskroom/miniconda/base/envs/ManyFEWS

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,0 +1,84 @@
+version: '3.6'
+
+services:
+
+  # RabbitMQ message queue service for Celery task broker
+  rabbitmq:
+    image: rabbitmq:3.10
+    container_name: ${COMPOSE_PROJECT_NAME}_rabbitmq
+    restart: unless-stopped
+    environment:
+      RABBITMQ_DEFAULT_USER: manyfews
+      RABBITMQ_DEFAULT_PASS: ${rabbitmq_password}
+    # ports:
+    #   - 5672:5672 # AMQP no TLS
+
+
+  # PostGIS server
+  postgres:
+    image: postgis/postgis:14-3.2
+    container_name: ${COMPOSE_PROJECT_NAME}_postgres
+    environment:
+      # must specify password for PG Docker container image, see: https://registry.hub.docker.com/_/postgres?tab=description&page=1&name=10
+      POSTGRES_DB: ${db_name}
+      POSTGRES_USER: ${db_user}
+      POSTGRES_PASSWORD: ${db_password}
+    ports:
+      - 5432:5432
+    # Needed because the postgres container does not provide a healthcheck
+    healthcheck:
+      test: pg_isready
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+
+  # Celery container runs from the same image as the django container below
+  celery:
+    container_name: ${COMPOSE_PROJECT_NAME}_celery
+    build:
+      context: .
+    user: celery:celery
+    restart: unless-stopped
+    env_file:
+      - .env
+    environment:
+      CELERY_BROKER_URL: "amqp://manyfews:${rabbitmq_password}@rabbitmq:5672/"
+    depends_on:
+      - "rabbitmq"
+    working_dir: /app/
+    command: celery -A manyfews worker --loglevel=INFO --scheduler django_celery_beat.schedulers:DatabaseScheduler
+
+
+  # Django container
+  #
+  # Run commands within the running container using:
+  # docker exec -it <container_id> /bin/bash
+  # Use `docker ps` to find the <container_id>
+  django:
+    container_name: ${COMPOSE_PROJECT_NAME}_django
+    build:
+      context: .
+    # restart: unless-stopped
+    env_file:
+      - .env
+    environment:
+      CELERY_BROKER_URL: "amqp://manyfews:${rabbitmq_password}@rabbitmq:5672/"
+
+    depends_on:
+      - "postgres"
+      - "rabbitmq"
+      - "celery"
+    ports:
+      - 8000:8000
+
+    entrypoint:
+      [ "bash", "--login", "-c", 
+        "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"]
+
+# You don't need this for production because the Dockerfile
+# does COPY . /app, but it can help for development!
+#    volumes:
+#      - "${PWD}/Data/:/app/Data"
+#      - "${PWD}/server/:/app/"
+

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,0 +1,14 @@
+#!/bin/bash --login
+# https://pythonspeed.com/articles/activate-conda-dockerfile/
+# The --login ensures the bash configuration is loaded,
+# enabling Conda.
+
+# Temporarily disable strict mode and activate conda:
+set +euo pipefail
+conda activate ManyFEWS
+
+# Re-enable strict mode:
+set -euo pipefail
+
+# exec the final command from script parameters:
+exec $*

